flow/connectors/mysql/validate.go

```diff
@@ -1,14 +1,172 @@
 package connmysql
 
+
 import (
 	"context"
+	"database/sql"
 	"errors"
+	"fmt"
+	"strings"
 
 	"github.com/go-mysql-org/go-mysql/mysql"
+	_ "github.com/go-sql-driver/mysql"
 
+	"github.com/PeerDB-io/peerdb/flow/connectors/utils"
 	"github.com/PeerDB-io/peerdb/flow/generated/protos"
+	"github.com/PeerDB-io/peerdb/flow/shared"
 )
 
+type MySQLConnector struct {
+	conn *sql.DB
+	config *MySQLConfig
+}
+
+func (c *MySQLConnector) CheckSourceTables(ctx context.Context, tableNames []*utils.SchemaTable) error {
+	if c.conn == nil {
+		return errors.New("check tables: conn is nil")
+	}
+
+	for _, parsedTable := range tableNames {
+		query := fmt.Sprintf("SELECT 1 FROM `%s`.`%s` LIMIT 1", parsedTable.Schema, parsedTable.Table)
+		_, err := c.conn.QueryContext(ctx, query)
+		if err != nil {
+			return fmt.Errorf("error checking table %s.%s: %v", parsedTable.Schema, parsedTable.Table, err)
+		}
+	}
+	return nil
+}
+
+func (c *MySQLConnector) CheckReplicationPermissions(ctx context.Context) error {
+	if c.conn == nil {
+		return errors.New("check replication permissions: conn is nil")
+	}
+
+	var replicationPrivilege string
+	err := c.conn.QueryRowContext(ctx, "SHOW GRANTS FOR CURRENT_USER()").Scan(&replicationPrivilege)
```

Comment:
it's more robust to parse this into a string array and then check to avoid matching partial permission names later